Services/src/main/java/ev/projects/services/CaseService.java

```diff
@@ -4,28 +4,58 @@
 import ev.projects.repositories.ICaseRepository;
 import org.springframework.beans.factory.annotation.Autowired;
 import org.springframework.context.annotation.ComponentScan;
+import org.springframework.security.core.Authentication;
+import org.springframework.security.core.context.SecurityContextHolder;
 import org.springframework.stereotype.Service;
 
+import java.util.Date;
 import java.util.List;
 import java.util.Optional;
 
 @Service
 public class CaseService implements ICaseService {
 
     private ICaseRepository caseRepository;
+    private IUserService userService;
 
     @Autowired
-    public CaseService(ICaseRepository caseRepository) {
+    public CaseService(ICaseRepository caseRepository, IUserService userService) {
         this.caseRepository = caseRepository;
+        this.userService = userService;
     }
 
     @Override
     public List<Case> getAll() {
-        return caseRepository.getAll();
+        return caseRepository.findAll();
     }
 
     @Override
-    public Optional<Case> getAllById(Long ID) {
-        return caseRepository.getById(ID);
+    public Optional<Case> getById(Long ID) {
+        return caseRepository.findById(ID);
     }
+
+    @Override
+    public Case add(Case aCase) {
+        Authentication auth = SecurityContextHolder.getContext().getAuthentication();
+        return userService.getByUsername(auth.getName()).map(user -> {
+            aCase.setCreatorUser(user);
+            aCase.setCreationDate(new Date());
+            return caseRepository.save(aCase);
+        }).orElse(null);
+    }
+
+    @Override
+    public boolean update(Case aCase) {
+        return caseRepository.findById(aCase.getID()).map(c -> {
```

Comment:
The idea was to not allow user call PUT (which calls update) when they POST the data.
They should call POST. So basically just to use the endpoints for their purpose.
Of course, maybe it is not needed.
Will implement this as noticed then.

Reply:
I tried `caseRepository.save()` as suggested and the update test failed. I think the reason is that if you save untracked object to db, new one is created no matter the ID. So findById is still probably needed.
However, I added the functionality where if Id is not found, update acts like add.
Suggestion in bee359711b1fde068dc4647167cb67883fdd0c16. Should I look for other solutions or will it be OK?

Reply:
Well, usually rest service accepts a DTO which is representation of an entity specifically for mapping it from json and then to an entity object. Kind of what you are doing by copying fields to entity. 
Speaking of untracked changes it is very strange. There should not be any issues with that if id is present. Not sure why this can happen, but whatever. May be it is worth adding custom update method to repository, with custom sql. Having 2 database operations instead of 1 is never a good solution.

Reply:
I create fake parents with IDs when I create an object (for example like this at EntitiesCreator (test util class)):
`if(caseID > 0) {
            Case owningCase = new Case();
            owningCase.setDocuments(List.of(document));
            owningCase.setID(caseID);
            document.setOwningCase(owningCase);
        }`
But that does not work with update. I think JPA probably for some reason does not recognize entry in db with the same ID as the same object.

I think having a custom sql is kind of problematic, because if you add new fields to Entity, the custom INSERT then should be updated right?

What about my suggested alternative in 12dd92b? It does not do the thing in one step but it does not fetch from db also, as stated in: https://stackoverflow.com/questions/39741102/how-to-beautifully-update-a-jpa-entity-in-spring-data

So the update now works with one db operation.

Does this suggestion fits you?

Reply:
All the tests passes with this alternative.

Reply:
If there are problems with persistence and takes too much time, leave it as it was.